bin/flutter.bat

```diff
@@ -166,20 +166,15 @@ GOTO :after_subroutine
 
 :after_subroutine
 
-CALL "%dart%" %FLUTTER_TOOL_ARGS% "%snapshot_path%" %*
-SET exit_code=%ERRORLEVEL%
-
-REM The VM exits with code 253 if the snapshot version is out-of-date.
-IF "%exit_code%" EQU "253" (
-  CALL "%dart%" --snapshot="%snapshot_path%" --packages="%flutter_tools_dir%\.packages" "%script_path%"
-  SET exit_code=%ERRORLEVEL%
-  IF "%exit_code%" EQU "253" (
-    ECHO Error: Unable to create dart snapshot for flutter tool.
-    EXIT /B %exit_code%
-  )
-  CALL "%dart%" %FLUTTER_TOOL_ARGS% "%snapshot_path%" %*
-  SET exit_code=%ERRORLEVEL%
-)
+REM Chaining the call to 'dart' and 'exit' with an ampersand ensures that
+REM Windows reads both commands into memory once before executing them. This
+REM avoids nasty errors that may otherwise occure when the dart command (e.g. as
+REM part of 'flutter upgrade') modifies this batch script while it is executing.
+REM
+REM Do not use the CALL command in the next line to execute Dart. CALL causes
```

Comment:
Whoa, who would have thought! Is this documented anywhere?

Reply:
I did not find any official documentation around this. There was one stack overflow answer explaining this behavior, but I can't find it now... :(